cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -2079,6 +2210,50 @@ Slice::Python::getAll(const ContainedPtr& definition)
     return all;
 }
 
+Slice::MetadataPtr
+Slice::Python::getSequenceMetadata(const SequencePtr& seq, const MetadataList& localMetadata)
+{
+    auto sequenceMetaData =
+        array{"python:list", "python:tuple", "python:array.array", "python:numpy.ndarray", "python:memoryview"};
+
+    // First check source metadata. For example, an operation parameter.
+    // If nothing was found, check the sequence itself.
+    for (const auto& metadataList : std::array{localMetadata, seq->getMetadata()})
+    {
+        for (const auto& metadata : metadataList)
+        {
+            if (find(sequenceMetaData.begin(), sequenceMetaData.end(), metadata->directive()) != sequenceMetaData.end())
+            {
+                return metadata;
+            }
+        }
+    }
+
+    return nullptr;
+}
+
+pair<string, string>
+Slice::Python::splitFQDN(const std::string& fqdn)
+{
+    size_t lastDot = fqdn.rfind('.');
+    if (lastDot != std::string::npos)
+    {
+        return {fqdn.substr(0, lastDot), fqdn.substr(lastDot + 1)};
+    }
+    return {"", fqdn};
+}
+
+pair<string, optional<string>>
+Slice::Python::splitMemoryviewArguments(const string& arguments)
+{
+    size_t pos = arguments.find(':');
+    if (pos != std::string::npos)
```

Comment:
Nor this. Obviously nothing _wrong_ with qualifying it, but we should be consistent.
```suggestion
    if (pos != string::npos)
```